spec/samples/unused_code.lua

```diff
@@ -21,3 +21,18 @@ y = 6
 local z = 5;
 (function() z = 4 end)()
 z = 6
+
+foo[foo] = 1
```

Comment:
This section makes my head spin just a bit. Perhaps it would be worth while adding some code comments to annotate which bits are expected to trip a warning and which bits are there to show that it isn't turning up false positives.